### PR TITLE
Feature: CUDA C++ continuity kernel implementation

### PR DESCRIPTION
**Pull request by drewsilcock**
**NOTE:** Branched from boundary condition and next kernel implementations, so those branches need to be merged first.

### CODE REVIEW

File: cuda/cuda_nemolite2d.cu [hunk 1]
```diff
@@ -319,6 +376,193 @@ cuda_initialise_grid_()
...
+  };
+}
+
+void
+cuda_continuity_()
+{
+  const int jpi = model_params.jpi;
+  const int jpj = model_params.jpj;
+
+  k_continuity<<<jpi + 1, jpj + 1>>>(*simulation_vars.sshn,
```
Comment (ptheywood) [inline cuda/cuda_nemolite2d.cu:415]: 1D, potentially invalid launch. See comments on #25  / #26

File: cuda/cuda_nemolite2d.cu [hunk 2]
```diff
@@ -319,6 +376,193 @@ cuda_initialise_grid_()
...
+  // TODO:
+}
+
+void
+cuda_boundary_conditions_(wp_t rtime)
+{
+  const int jpi = model_params.jpi;
+  const int jpj = model_params.jpj;
+
+  k_boundary_conditions<<<jpi + 1, jpj + 1>>>(rtime,
```
Comment (ptheywood) [inline cuda/cuda_nemolite2d.cu:447]: 1D, potentially invalid launch. See comments on #25  / #26

File: cuda/cuda_nemolite2d.cu [hunk 3]
```diff
@@ -319,6 +376,193 @@ cuda_initialise_grid_()
...
+  simulation_vars.va->set_device_data_ptr(va_buffered_data);
+}
+
+void
+cuda_next_()
+{
+  const int jpi = model_params.jpi;
+  const int jpj = model_params.jpj;
+
+  k_next<<<jpi + 1, jpj + 1>>>(*simulation_vars.sshn,
```
Comment (ptheywood) [inline cuda/cuda_nemolite2d.cu:491]: 1D, potentially invalid launch. See comments on #25  / #26

**Review by ptheywood — CHANGES_REQUESTED** [review]
1D, potentially invalid launches as with the other PRs. 

Code otherwise looks sensible.

**Comment by drewsilcock** [comment]
All these different PRs are getting a bit confusing now so I'm going to consolidate them into a single one.

**Pull request closed without merging**